app/scripts/views/connect_another_device.js

```diff
@@ -0,0 +1,269 @@
+      const isOther = ! isAndroid && ! isIos;
+
+      let visibleAreaSuffix;
+      if (canSignIn) {
+        this.notifier.trigger('connectAnotherDevice.signin.eligible');
+
+        if (isFirefoxAndroid) {
+          visibleAreaSuffix = 'signin_from.fennec';
+        } else {
+          visibleAreaSuffix = 'signin_from.desktop';
+        }
+      } else {
+        this.notifier.trigger('connectAnotherDevice.signin.ineligible');
+
+        if (isFirefoxIos) {
+          visibleAreaSuffix = 'signin_from.fxios';
+        } else if (isOtherIos) {
+          visibleAreaSuffix = 'install_from.other_ios';
+        } else if (isFirefoxAndroid) {
+          visibleAreaSuffix = 'install_from.fennec';
+        } else if (isOtherAndroid) {
+          visibleAreaSuffix = 'install_from.other_android';
+        } else if (isOther) {
+          visibleAreaSuffix = 'install_from.other';
+        }
+      }
+
+      if (visibleAreaSuffix) {
+        this.notifier.trigger(`connectAnotherDevice.${visibleAreaSuffix}`);
+      }
+
+      return {
+        canSignIn,
+        email,
+        escapedSignInUrl,
+        isFirefoxAndroid,
+        isFirefoxIos,
+        isOther,
+        isOtherAndroid,
+        isOtherIos
+      };
+    },
+
+    /**
+     * Get the user-agent string. For functional testing
+     * purposes, first attempts to fetch a UA string from the
+     * `forceUA` query parameter, if that is not found, use
+     * the browser's.
+     *
+     * @returns {String}
+     * @private
+     */
+    _getUserAgentString () {
+      return this.getSearchParam('forceUA') || this.window.navigator.userAgent;
+    },
+
+    /**
+     * Get a user-agent parser instance.
+     *
+     * @returns {Object}
+     * @private
+     */
+    _getUap () {
+      if (! this._uap) {
+        const userAgent = this._getUserAgentString();
+        this._uap = new UserAgent(userAgent);
+      }
+      return this._uap;
+    },
+
+    /**
+     * Check if the current user can sign in.
+     *
+     * @returns {Boolean}
+     * @private
+     */
+    _canSignIn () {
+      // Only users that are not signed in can do so.
+      return ! this.model.get('isSignedIn') &&
+               this._hasWebChannelSupport();
+    },
+
+    /**
+     * Check if the current browser has web channel support.
+     *
+     * @returns {Boolean}
+     * @private
+     */
+    _hasWebChannelSupport () {
+      const uap = this._getUap();
+
+        // All Foxes >= 40 except iOS can sign in.
+      return uap.isFirefox() &&
+           ! uap.isIos() &&
+             uap.browser.version >= 40;
+    },
+
+    /**
+     * Return the sign in context that can be used to sign in
+     * to Sync. Assumes the context is only used if the user
+     * can actually sign in to Sync.
+     *
+     * @returns {String}
+     * @private
+     */
+    _getSignInContext() {
+      if (this._getUap().isAndroid()) {
+        return Constants.FX_FENNEC_V1_CONTEXT;
+      } else {
```

Comment:
Add a `isFirefox() && ! isFirefoxIos()` check here to avoid making assumptions about where the output of this function will be used.